courses/fundamentals_of_ada/060_record_types.rst

```diff
@@ -621,121 +547,119 @@ Quiz
+   + All object **still** share the same type
 
-* Variant record offers a kind of storage overlaying
+* Kind of **storage overlay**
 
-   + Same storage might be used for one variant at one time, and then for another variant later
-   + Language issue: Ensure this does not provide loophole from type checking
+   + Similar to :C:`union` in C
+   + But preserves **type checking**
+   + And object size **depends** on discriminant
 
-      * Neither Pascal nor C avoids this loophole
+* Aggregate assignment is allowed
 
--------------------------------------
-Discriminant in Ada Variant Records
--------------------------------------
+---------------
+Discriminants
+---------------
 
 .. code:: Ada
 
-  type Person_Tag is (Student, Faculty);
-  type Person (Tag : Person_Tag) is record
+  type Person_Group is (Student, Faculty);
+  type Person (Group : Person_Group) is record
      Name : String (1 .. 10);
-     case Tag is
+     case Group is
         when Student => -- 1st variant
            Gpa  : Float range 0.0 .. 4.0;
-           Year : Integer range 1 .. 4;
         when Faculty => -- 2nd variant
            Pubs : Integer;
      end case;
   end record;
 
-* Variant record type contains a special field (*discriminant*) whose value indicates which variant is present
-* When a field in a variant is selected, run-time check ensures that discriminant value is consistent with the selection
+* :ada:`Group` is the **discriminant**
+* Run-time check for component **consistency**
 
-   + If you could store into `Pubs` but read `GPA`, type safety would not be guaranteed
+   + eg :ada:`A_Person.Pubs := 1` checks :ada:`A_Person.Group = Faculty`
+   + :ada:`Constraint_Error` if check fails
 
-* Ada prevents this type of access
+* Discriminant is **constant**
 
-   + Discriminant (Tag) established when object of type Person created
-   + Run-time check verifies that field selected from variant is consistent with discriminant value
+   + Unless object is **mutable**
 
-      * Constraint_Error raised if the check fails
+------------------------------
+Mutable Discriminated Record
+------------------------------
+
+* When discriminant has a **default value**
+
+   + Any object instanciated **without** a discriminant value is **mutable**
+
+* Mutable records have **variable** discriminants
+* Use **same** storage for **several** variant
+
+.. code:: Ada
 
-* Can only read discriminant (as any other field), not write
+  -- Potentially mutable
+  type Person (Group : Person_Group := Student) is record
 
-      * Aggregate assignment is allowed
+  -- Uses default value: mutable
+  S : Person;
+  -- Specifies the value: not mutable
+  S2 : Person (Group => Student);
+  ...
+  S := (Group => Student, Gpa => 0.0);
+  S := (Group => Faculty, Pubs => 10);
 
 -----------
 Semantics
 -----------
 
-* Variable of type `Person` is constrained by value of discriminant supplied at object declaration
+* :ada:`Person` objects are **constrained** by their discriminant
```

Comment:
This section is NOT TRUE for discriminants with defaults

Reply:
Rewrote it